features/sharding/core/src/main/java/org/apache/shardingsphere/sharding/merge/dql/groupby/aggregation/GroupConcatAggregationUnit.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.shardingsphere.sharding.merge.dql.groupby.aggregation;
+
+import lombok.NoArgsConstructor;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@NoArgsConstructor
+public class GroupConcatAggregationUnit implements AggregationUnit {
+    
+    private static final String DEFAULT_SEPARATOR = ",";
+    
+    private final Collection<Comparable<?>> values = new ArrayList<>();
+    
+    private String separator;
+    
+    public GroupConcatAggregationUnit(final String separator) {
+        this.separator = separator;
+    }
+    
+    @Override
+    public void merge(final List<Comparable<?>> values) {
+        if (null == values || null == values.get(0)) {
+            return;
+        }
+        this.values.add(values.get(0));
+    }
+    
+    @Override
+    public Comparable<?> getResult() {
+        if (null == separator) {
+            separator = DEFAULT_SEPARATOR;
+        }
+        return values.stream().map(Object::toString).collect(Collectors.joining(separator));
```

Comment:
Can you cast value to string when execute merge method? This way we can reduce one loop call.